app/src/store/features/variables/reducers.ts

```diff
@@ -15,9 +15,10 @@ const addNewEnvironment = (
   action: PayloadAction<{
     id: string;
     name: string;
+    ownerId: string;
   }>
 ) => {
-  state.environments[action.payload.id] = {
+  state.environments[action.payload.ownerId][action.payload.id] = {
```

Comment:
@nafees87n  why `state.environments.<ownerId>.<environmentId>` ? why not `state.environments..<environmentId>` directly?

Reply:
Changed the data structure [here](https://github.com/requestly/requestly/pull/2533/files#diff-43dcf15560a89a7da8a241595dff94c6f78547275c538f81b263aee10e909930R5-R7)

All the user's environment will be cached in redux lazily as user keeps on switching across workspaces